space_packet_parser/xtcedef.py

```diff
@@ -1180,38 +1230,43 @@ def parse_value(self, packet_data: PacketData, parsed_data: dict, **kwargs) -> T
 class StringDataEncoding(DataEncoding):
     """<xtce:StringDataEncoding>"""
 
-    def __init__(self, encoding: str = 'utf-8',
+    _supported_encodings = ('US-ASCII', 'ISO-8859-1', 'Windows-1252', 'UTF-8', 'UTF-16',
+                            'UTF-16LE', 'UTF-16BE', 'UTF-32', 'UTF-32LE', 'UTF-32BE')
+
+    def __init__(self, encoding: str = 'UTF-8',
+                 byte_order: Optional[str] = None,
                  termination_character: Optional[str] = None,
                  fixed_length: Optional[int] = None,
                  leading_length_size: Optional[int] = None,
                  dynamic_length_reference: Optional[str] = None,
-                 use_calibrated_value: bool = True,
+                 use_calibrated_value: Optional[bool] = True,
                  discrete_lookup_length: Optional[List[DiscreteLookup]] = None,
                  length_linear_adjuster: Optional[callable] = None):
         """Constructor
         Only one of termination_character, fixed_length, or leading_length_size should be set. Setting more than one
         is nonsensical.
 
-        TODO: implement ByteOrderList to inform endianness.
-         This can also relax the requirements on the encoding spec since utf-16-le is redundant if endianness
-         comes from the ByteOrderList
-
         Parameters
         ----------
         encoding : str
-            One of 'utf-8', 'utf-16-le', or 'utf-16-be'. Describes how to read the characters in the string.
+            One of the XTCE-supported encodings, e.g. UTF-8, UTF-16LE.
```

Comment:
Maybe inject the encodings here?
```suggestion
            One of the XTCE-supported encodings, {self._supported_encodings}
```

Reply:
This was throwing pylint and pycodestyle errors but I can suppress them.